### ISSUE TITLE
Formatter setting ignored

### ISSUE THREAD

**Issue by georgefst** [haskell/vscode-haskell#246]
With version 0.1, I'm finding that, regardless of what I set, Ormolu is used as the formatting provider.

The following is logged when I try to change the setting through the GUI:
```
[Error - 01:17:57] haskell-lsp:configuration parse error. NotificationMessage {_jsonrpc = "2.0", _method = WorkspaceDidChangeConfiguration, _params = DidChangeConfigurationParams {_settings = Object (fromList [("haskell",Object (fromList [("logFile",String ""),("hlintOn",Bool True),("formatOnImportOn",Bool True),("enable",Bool True),("indentationRules",Object (fromList [("enabled",Bool True)])),("liquidOn",Bool False),("languageServerVariant",String "haskell-language-server"),("serverExecutablePath",String "${HOME}/.cabal/bin/haskell-language-server"),("diagnosticsOnChange",Bool True),("completionSnippetsOn",Bool True),("maxNumberOfProblems",Number 100.0),("formattingProvider",String "brittany"),("trace",Object (fromList [("server",String "off")]))]))])}} "key \"languageServerHaskell\" not found"
```

**Comment by alanz** [comment]
Which backend are you using? This only works with haskell language server, to my knowledge

**Comment by konn** [comment]
Same here, with `haskell-language-server` 0.2.1 backend, automatically downloaded by the extension, with `stylish-haskell` configured as a formatter.
Actually, it seems that no formattings are taken place.

In OUTPUT pane, a similar message is displayed when I change the formatter after the server launched:

```
[Error - 16:49:21] haskell-lsp:configuration parse error. NotificationMessage {_jsonrpc = "2.0", _method = WorkspaceDidChangeConfiguration, _params = DidChangeConfigurationParams {_settings = Object (fromList [("haskell",Object (fromList [("hlint",Object (fromList [("logLevel",String "info"),("executablePath",String "/Users/hiromi/.local/bin/hlint")])),("logFile",String ""),("hlintOn",Bool True),("formatOnImportOn",Bool True),("enable",Bool True),("indentationRules",Object (fromList [("enabled",Bool True)])),("liquidOn",Bool False),("languageServerVariant",String "haskell-language-server"),("serverExecutablePath",String ""),("diagnosticsOnChange",Bool True),("completionSnippetsOn",Bool True),("maxNumberOfProblems",Number 100.0),("formattingProvider",String "floskell"),("trace",Object (fromList [("server",String "off")]))]))])}} "key \"languageServerHaskell\" not found"
```

Perhaps JSON schema changed in 0.2.1?

**Comment by georgefst** [comment]
> Which backend are you using?

HLS. Both with the binary auto-downloaded by the extension, and with one I built myself that worked correctly with the previous version.

**Comment by jneira** [comment]
https://github.com/haskell/haskell-language-server/issues/226 will fix this one

**Comment by bubba** [comment]
Whoops. I renamed the configuration section to be inline with the other language servers (i.e. "rust.blah", so "langaugeServerHaskell.blah" => "haskell.blah") but completely forgot to actually update it in hls. Fix is here https://github.com/haskell/haskell-language-server/pull/227

**Comment by bubba** [comment]
HIE fix is https://github.com/haskell/haskell-ide-engine/pull/1794

**Comment by bubba** [comment]
Fixed in hls 0.2.2
